### PR TITLE
Enforce Sinon Sandbox and update non-sandbox cases.

### PR DESCRIPTION
**Pull request by mkhatib**

### CODE REVIEW

File: build-system/tasks/presubmit-checks.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ var forbiddenTerms = {
   'DO NOT SUBMIT': '',
   'describe\\.only': '',
   'it\\.only': '',
-  'sinon\\.(spy|stub|mock)\\(\\w[^)]*\\)': {
+  'sinon\\.(spy|stub|mock)\\((\\w[^)])*\\)': {
```
Comment (jridgewell) [inline build-system/tasks/presubmit-checks.js]: We can simplify this to `sinon\\.(spy|stub|mock)\\(`.

**Comment by jridgewell** [comment]
I'm ambivalent to this change. I originally left anonymous `sinon.spy()` alone because they do not modify anything, and declaring them inside the sandbox is essentially a more expensive noop. Other than that, I have one comment, then LGTM.

**Comment by mkhatib** [comment]
Yeah I had a feeling that this was the case. I am happy to leave the anonymous alone, though I am worried it would encourage people to use sinon directly without a sandbox with non-anon spys/stubs...

I think forcing a sandbox just gets us used to always using a sandbox.

File: build-system/tasks/presubmit-checks.js (hunk 1 above)
Comment (mkhatib) [inline build-system/tasks/presubmit-checks.js]: Done.

**Comment by jridgewell** [comment]
Good with me.

**Pull request merged**